File: go-sdk/kestra_api_client/security_integrations_api.go
package kestra_api_client

import "context"

// SecurityIntegrationsAPI covers /api/v1/{tenant}/security-integrations (SCIM).
// The controller requires instance-owner access and the FEATURE_SCIM license
// feature, so routes answer 403 otherwise.
type SecurityIntegrationsAPI struct {
	baseAPI
}

// ListSecurityIntegrations returns every security integration in an envelope
// {total, results}. Deprecated server-side in favour of SearchSecurityIntegrations.
// Backs GET /api/v1/{tenant}/security-integrations.
func (a *SecurityIntegrationsAPI) ListSecurityIntegrations(ctx context.Context, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "GET", tenantPath(tenant, "security-integrations"), nil, nil)
}

// SearchSecurityIntegrations returns a page of security integrations. Backs GET
// /api/v1/{tenant}/security-integrations/search.
func (a *SecurityIntegrationsAPI) SearchSecurityIntegrations(ctx context.Context, tenant string, page, size *int, sort []string, filters []SearchFilter) (map[string]interface{}, error) {
	params := buildQueryParams("page", page, "size", size)
	appendRepeatedParam(params, "sort", sort)
	if err := appendFilterParams(params, filters); err != nil {
		return nil, err
	}
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "GET", tenantPath(tenant, "security-integrations", "search"), nil, params)
}

// SecurityIntegration returns a single security integration by id. Backs GET
// /api/v1/{tenant}/security-integrations/{id}.
func (a *SecurityIntegrationsAPI) SecurityIntegration(ctx context.Context, id, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "GET", tenantPath(tenant, "security-integrations", id), nil, nil)
}

// CreateSecurityIntegration creates a security integration. `body` is a
// CreateSecurityIntegrationRequest ({name, description, type}); the response
// carries the SCIM bearer token once, at creation. Backs POST
// /api/v1/{tenant}/security-integrations.
func (a *SecurityIntegrationsAPI) CreateSecurityIntegration(ctx context.Context, tenant string, body interface{}) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "POST", tenantPath(tenant, "security-integrations"), body, nil)
}

// DeleteSecurityIntegration deletes a security integration. Backs DELETE
// /api/v1/{tenant}/security-integrations/{id}.
func (a *SecurityIntegrationsAPI) DeleteSecurityIntegration(ctx context.Context, id, tenant string) error {
	return a.doVoidJSON(ctx, "DELETE", tenantPath(tenant, "security-integrations", id), nil, nil)
}

// EnableSecurityIntegration enables a security integration. Backs POST
// /api/v1/{tenant}/security-integrations/{id}/enable.
func (a *SecurityIntegrationsAPI) EnableSecurityIntegration(ctx context.Context, id, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "POST", tenantPath(tenant, "security-integrations", id, "enable"), nil, nil)
}

// DisableSecurityIntegration disables a security integration. Backs POST
// /api/v1/{tenant}/security-integrations/{id}/disable.
func (a *SecurityIntegrationsAPI) DisableSecurityIntegration(ctx context.Context, id, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "POST", tenantPath(tenant, "security-integrations", id, "disable"), nil, nil)
}

File: python/python-sdk/kestrapy/api/security_integrations_api.py
from typing import Any, Dict, List, Optional

from kestrapy.base_api import BaseApi
from kestrapy.models.create_security_integration_request import CreateSecurityIntegrationRequest
from kestrapy.models.paged_results_security_integration import PagedResultsSecurityIntegration
from kestrapy.models.query_filter import QueryFilter
from kestrapy.models.security_integration import SecurityIntegration


class SecurityIntegrationsApi(BaseApi):
    """Security integrations / SCIM (`/api/v1/{tenant}/security-integrations`).

    The controller requires instance-owner access and the ``FEATURE_SCIM``
    licence feature, so every route answers 403 when the feature is disabled.
    """

    def list_security_integrations(self, tenant: str) -> PagedResultsSecurityIntegration:
        """List every security integration in an envelope ``{total, results}``.
        Deprecated server-side in favour of :meth:`search_security_integrations`."""
        path = self._tenant_path(tenant, "security-integrations")
        return self._json_request("GET", path, PagedResultsSecurityIntegration)

    def search_security_integrations(
        self,
        tenant: str,
        page: Optional[int] = None,
        size: Optional[int] = None,
        sort: Optional[List[str]] = None,
        filters: Optional[List[QueryFilter]] = None,
    ) -> PagedResultsSecurityIntegration:
        path = self._tenant_path(tenant, "security-integrations", "search")
        params = list(self._build_query_params(page=page, size=size).items())
        self._append_repeated_param(params, "sort", sort)
        self._append_filter_params(params, filters)
        return self._json_request("GET", path, PagedResultsSecurityIntegration, params=params)

    def security_integration(self, id: str, tenant: str) -> SecurityIntegration:
        path = self._tenant_path(tenant, "security-integrations", id)
        return self._json_request("GET", path, SecurityIntegration)

    def create_security_integration(
        self, tenant: str, body: CreateSecurityIntegrationRequest
    ) -> Dict[str, Any]:
        """Create a security integration. The response carries the SCIM bearer
        token once, at creation — which is not modelled — so a raw dict is
        returned to preserve it."""
        path = self._tenant_path(tenant, "security-integrations")
        return self._raw_json_request("POST", path, body=body)

    def delete_security_integration(self, id: str, tenant: str) -> None:
        path = self._tenant_path(tenant, "security-integrations", id)
        self._void_request("DELETE", path)

    def enable_security_integration(self, id: str, tenant: str) -> SecurityIntegration:
        path = self._tenant_path(tenant, "security-integrations", id, "enable")
        return self._json_request("POST", path, SecurityIntegration)

    def disable_security_integration(self, id: str, tenant: str) -> SecurityIntegration:
        path = self._tenant_path(tenant, "security-integrations", id, "disable")
        return self._json_request("POST", path, SecurityIntegration)
